dashboard-frontend/src/pages/AccountsPage/InputTypeformKey.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { ChangeEventHandler } from 'react';
+import { classNames } from '../../helpers/strings';
+
+const InputTypeformKey: React.FC<any> = ({
+  error,
+  handleCredentialChange,
+  index,
+  entity,
+  typeformkey,
+}: {
+  index: number;
+  error: string | undefined;
+  handleCredentialChange: ChangeEventHandler;
+  entity: string;
+  typeformkey: string;
+}) => (
+  <React.Fragment>
+    <div className="flex flex-col mb-4">
+      <label className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300">
+        Credential ID
+      </label>
+      <input
+        id={`${index}`}
+        name="identifier"
+        type="text"
+        className={classNames(
+          'block w-full shadow-sm sm:text-sm rounded-md',
+          (error ?? '').trim() !== ''
+            ? 'focus:ring-red-500 focus:border-red-500 border-red-300 text-red-900 pr-10'
+            : 'focus:ring-indigo-500 focus:border-indigo-500 border-gray-300'
+        )}
+        placeholder="Name of Credential"
+        value={entity}
+        disabled
+      />
+    </div>
+    <div className="flex flex-col mb-4">
+      <label className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300">
+        Key
+      </label>
+      <input
+        id="key"
+        name="key"
+        type="password"
+        className={classNames(
+          'block w-full shadow-sm sm:text-sm rounded-md',
+          (error ?? '').trim() !== ''
+            ? 'focus:ring-red-500 focus:border-red-500 border-red-300 text-red-900 pr-10'
+            : 'focus:ring-indigo-500 focus:border-indigo-500 border-gray-300'
+        )}
+        data-testid={`input-typeform-${index}`}
+        placeholder="Enter API Key"
+        onChange={handleCredentialChange}
+        value={typeformkey}
+        required
+      />
+    </div>
+  </React.Fragment>
+);
+
+export default InputTypeformKey;
```

Comment:
Nice that these two components are so similar as they can definitely be dried out at a later date into one!

Reply:
Yeah I really dont like it but kept it :sweat_smile:

Reply:
Actually solved this